tmt/steps/report/polarion.py

```diff
@@ -0,0 +1,121 @@
+        """ Prepare command line options """
+        options = super().options(how)
+        options[:0] = [
+            click.option(
+                '--file', metavar='FILE', help='Path to the file to store xUnit in'),
+            click.option(
+                '--upload / --no-upload', default=True, show_default=True,
+                help="Whether to upload results to Polarion"),
+            click.option(
+                '--project-id', required=True, help='Use specific Polarion project ID'),
+            click.option(
+                '--testrun-title', help='Use specific TestRun title')
+            ]
+        return options
+
+    def go(self, *args: Any, **kwargs: Any) -> None:
+        """ Go through executed tests and report into Polarion """
+        super().go()
+
+        from tmt.export import get_polarion_ids, import_polarion
+        import_polarion()
+        from tmt.export import PolarionWorkItem
+        assert PolarionWorkItem
+
+        title = self.opt(
+            'testrun_title',
+            self.step.plan.name.rsplit('/', 1)[1] +
+            datetime.datetime.now().strftime("%Y%m%d%H%M%S"))
+        title = title.replace('-', '_')
+        project_id = self.opt('project-id')
+        upload = self.opt('upload')
+
+        junit_suite = make_junit_xml(self)
+        xml_tree = ET.fromstring(junit_suite.to_xml_string([junit_suite]))
+
+        properties = {
+            'polarion-project-id': project_id,
+            'polarion-user-id': PolarionWorkItem._session.user_id,
+            'polarion-testrun-id': title,
+            'polarion-project-span-ids': project_id}
+        testsuites_properties = ET.SubElement(xml_tree, 'properties')
+        for name, value in properties.items():
+            ET.SubElement(testsuites_properties, 'property', attrib={
+                'name': name, 'value': value})
+
+        testsuite = xml_tree.find('testsuite')
+        project_span_ids = xml_tree.find(
+            '*property[@name="polarion-project-span-ids"]')
+
+        for result in self.step.plan.execute.results():
+            if not result.id:
+                raise tmt.utils.ReportError(
+                    f"Test Case {result.name} is not exported to Polarion, "
+                    "please run 'tmt tests export --how polarion' on it")
+            work_item_id, test_project_id = get_polarion_ids(
+                PolarionWorkItem.query(
+                    result.id, fields=['work_item_id', 'project_id']))
+
+            if test_project_id not in project_span_ids.attrib['value']:
+                project_span_ids.attrib['value'] += f',{test_project_id}'
+
+            test_properties = {
+                'polarion-testcase-id': work_item_id,
+                'polarion-testcase-project-id': test_project_id}
+
+            test_case = testsuite.find(f"*[@name='{result.name}']")
+            properties_elem = ET.SubElement(test_case, 'properties')
```

Comment:
`tmt/steps/report/polarion.py:91: error: Argument 1 to "SubElement" has incompatible type "Union[Element, None, Any]"; expected "Element"  [arg-type]`

Same situation: `test_case` comes from `testsuite.find(f"*[@name='{result.name}']")`, may be `None`. Also, thanks to previous error, `testsuite` itself may be `None`, and `None.find()` is unknown, adding `Any` to the mix of possible types of `test_case`.